datasets/plugins/batch/batch_dataset_plugin.py

```diff
@@ -0,0 +1,173 @@
+from pathlib import Path
+from typing import TYPE_CHECKING, Iterable, Optional, Tuple, Union
+
+import pandas
+import pandas as pd
+
+from datasets import Mode
+from datasets.context import Context
+from datasets.dataset_plugin import DatasetPlugin
+
+
+if TYPE_CHECKING:
+    import dask.dataframe as dd
+    import pyspark
+
+
+class InvalidOperationException(Exception):
+    pass
+
+
+@DatasetPlugin.register_plugin(constructor_keys={"name"}, context=Context.BATCH)
+class BatchDatasetPlugin(DatasetPlugin):
+    """
+    This is the default plugin for the BATCH execution context.
+    """
+
+    _dataset_path_func: callable = None
```

Comment:
Minor: Seems like in some places in the code you use the method `callable`, other times you import from `typing` and use `Callable`.